traffic_ops/testing/api/v4/parameters_test.go

```diff
@@ -231,3 +340,204 @@ func DeleteTestParameter(t *testing.T, pl tc.Parameter) {
+				}
+			} else {
+				t.Error("Error in getting parameter by limit and offset")
+			}
+
+			qparams = url.Values{}
+			qparams.Set("orderby", "id")
+			qparams.Set("limit", "1")
+			qparams.Set("page", "2")
+			parametersWithPage, _, err := TOSession.GetParameters(qparams, nil)
+			if err == nil {
+				if !reflect.DeepEqual(parameters[1:2], parametersWithPage) {
+					t.Error("expected GET Parameters with limit = 1, page = 2 to return second result")
+				}
+			} else {
+				t.Error("Error in getting parameters by limit and page")
+			}
+		} else {
+			t.Errorf("only one parameters found, so offset functionality can't test")
+		}
+	} else {
+		t.Errorf("No parameters found to check pagination")
+	}
+
+	qparams = url.Values{}
+	qparams.Set("limit", "-2")
+	_, _, err = TOSession.GetParameters(qparams, nil)
+	if err == nil {
+		t.Error("expected GET Parameters to return an error when limit is not bigger than -1")
+	} else if !strings.Contains(err.Error(), "must be bigger than -1") {
+		t.Errorf("expected GET Parameters to return an error for limit is not bigger than -1, actual error: " + err.Error())
+	}
+
+	qparams = url.Values{}
+	qparams.Set("limit", "1")
+	qparams.Set("offset", "0")
+	_, _, err = TOSession.GetParameters(qparams, nil)
+	if err == nil {
+		t.Error("expected GET Parameters to return an error when offset is not a positive integer")
+	} else if !strings.Contains(err.Error(), "must be a positive integer") {
+		t.Errorf("expected GET Parameters to return an error for offset is not a positive integer, actual error: " + err.Error())
+	}
+
+	qparams = url.Values{}
+	qparams.Set("limit", "1")
+	qparams.Set("page", "0")
+	_, _, err = TOSession.GetParameters(qparams, nil)
+	if err == nil {
+		t.Error("expected GET Parameters to return an error when page is not a positive integer")
+	} else if !strings.Contains(err.Error(), "must be a positive integer") {
+		t.Errorf("expected GET Parameters to return an error for page is not a positive integer, actual error: " + err.Error())
+	}
+}
+
+func GetTestParametersByConfigfile(t *testing.T) {
+	for _, parameters := range testData.Parameters {
+		resp, reqInf, err := TOSession.GetParametersByConfigFile(parameters.ConfigFile, nil)
+		if err != nil {
+			t.Errorf("cannot GET Parameter by Config File: %v - %v", parameters.ConfigFile, err)
+		}
+		if reqInf.StatusCode != http.StatusOK {
+			t.Errorf("Expected 200 status code, got %v", reqInf.StatusCode)
+		}
+		if len(resp) <= 0 {
+			t.Errorf("No data available for Get Parameters by Config file")
+		}
+	}
+}
+
+func GetTestParametersByValue(t *testing.T) {
+	for _, parameters := range testData.Parameters {
+		_, reqInf, err := TOSession.GetParametersByValue(parameters.Value, nil)
+		if err != nil {
+			t.Errorf("cannot GET Parameter by Value: %v - %v", parameters.Value, err)
+		}
+		if reqInf.StatusCode != http.StatusOK {
+			t.Errorf("Expected 200 status code, got %v", reqInf.StatusCode)
+		}
+	}
+}
```

Comment:
How come this test doesn't look for `len(resp) <=0` like the ones above and below?

Reply:
added that validation also. I skipped it because the value can be null.

Reply:
That shouldn't be possible - unless there are Parameters in the testing data that aren't supposed to be created successfully, but if that were true, then I would expect the CreateMultipleParameters method's test to fail.

Reply:
Before executing the CreateMultipleParameters, I am deleting all the parameters that have created before.